insights/parsers/netstat.py

```diff
@@ -525,12 +525,19 @@ def rows_by(self, section_id, search_dict):
             the original line of input that produced this row.
         """
         results = []
+        if search_dict == {}:
+            return results
         for line, row in enumerate(self.datalist[section_id]):
+            # Row must match all criteria, so start with true and set to false
+            # if we find a non-match.
+            found = True
             for key, value in search_dict.iteritems():
-                if key in row and row[key] == value:
-                    copy = dict(row)
-                    copy['raw line'] = self.lines[section_id][line]
-                    results.append(copy)
+                if not (key in row and row[key] == value):
+                    found = False
+            if found:
+                copy = dict(row)
```

Comment:
@PaulWay you might consider optimizing this a bit by replacing 533-537 with this:

```python
if all([k in row and row[k] == v for k, v in search_dict.iteritems()]):
```